src/backer/cardaning.py

```diff
@@ -59,44 +65,104 @@ def __init__(self, name='backer', hab=None, ks=None):
-                self.pending_kel = bytes(self.pending_kel)
+                kel_data = bytes(kel)
                 # Chunk size
                 # bytearrays is not accept
-                value = [self.pending_kel[i:i + 64] for i in range(0, len(self.pending_kel), 64)]
+                value = [kel_data[i:i + 64] for i in range(0, len(kel_data), 64)]
 
                 # Metadata. accept int key type
                 builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
 
                 signed_tx = builder.build_and_sign([self.payment_signing_key],
                                                 change_address=self.spending_addr,
                                                 merge_change=True)
+                
                 # Submit transaction
-                self.context.submit_tx(signed_tx.to_cbor())  
-                self.pending_kel = bytearray()
-        except Exception as e:    
-            logger.critical(f"Error: {e}")
+                self.context.submit_tx(signed_tx.to_cbor())
+                transId = str(signed_tx.transaction_body.inputs[0].transaction_id)
+                trans = {
+                    "id": transId,
+                    "kel": kel.decode('utf-8'),
+                    "publish_time": help.toIso8601()
+                }
+                self.keldbConfirming.pin(keys=transId, val=json.dumps(trans).encode('utf-8'))
+        except Exception as e:
+            logger.critical(f"Cannot submit transaction: {e}")
+
+    def emptyPendingKEL(self):
+        self.pending_kel = bytearray()
+
+    def flushQueue(self):
+        if self.pending_kel is not None and self.pending_kel != b"":
+            self.submitKelTx(self.pending_kel)        
+            self.pending_kel = bytearray()
+
+    def slotToTime(self, slot):
+        timeStamp = SHELLY_UNIX + slot
+        return datetime.fromtimestamp(timeStamp)
+
+    def getConfirmingTrans(self, transId):
+        return self.keldbConfirming.get(transId)
+
+    def confirmTrans(self):
+        try:
+            for keys, item in self.keldbConfirming.getItemIter():
+                if item is None:
+                    continue
+                
+                item = json.loads(item)
+                (blockSlot, blockHeight) = (item["block_slot"], item["block_height"]) if 'block_slot' in item.keys() else (0, 0)
+                publishTime = datetime.fromisoformat(item["publish_time"])
+                onChainTime = help.toIso8601(self.slotToTime(blockSlot))
+                onChainTime = datetime.fromisoformat(onChainTime).replace(tzinfo=timezone.utc)
+                confirmTime = datetime.fromisoformat(help.toIso8601())
+
+                # Check for confirmation
+                if self.tipHeight - int(blockHeight) >= TRANSACTION_DEEP and (onChainTime - publishTime).total_seconds() <= TRANSACTION_CONFIRMATION_TIMEOUT:
```

Comment:
I think lets just simplify this if statement and only have `if self.tipHeight - int(blockHeight):`. If it gets this deep, no need to re-submit. We just re-submit after a timeout in case it _never_ appears.